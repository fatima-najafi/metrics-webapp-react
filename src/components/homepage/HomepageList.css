.context {
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-content: center;
  width: 100%;
  height: 100%;
}

.search {
  display: flex;
  justify-content: center;
  align-content: center;
  width: 100%;
  height: 100%;
  margin-top: 20px;
}

.input {
  text-indent: 20px;
  font-family: "Poppins", sans-serif;
  border-radius: 10px;
  margin-top: 0;
  height: 40px;
  width: 300px;
  font-weight: bold;
  color: white;
  background-color: rgba(115, 34, 93, 0.66);
  border: 2px solid rgb(255, 255, 255);
}

::placeholder {
  color: rgb(255, 255, 255);
  font-weight: bold;
}

.input:focus {
  outline: none;
  box-shadow: none;
  border: 1px solid rgb(255, 255, 255);
}

.container {
  display: grid;
  grid-template-columns: repeat(2, 0.4fr);
  justify-content: center;
  align-content: center;
}

ul {
  list-style: none;
}

/* for desctop version */
@media only screen and (min-width: 768px) {
  .context {
    display: flex;
    justify-content: center;
    align-content: center;
    width: 100%;
    height: 100%;
  }

  .input {
    text-indent: 20px;
    font-family: "Poppins", sans-serif;
    border-radius: 10px;
    margin-top: 0;
    height: 40px;
    width: 600px;
    font-weight: bold;
    color: white;
    background-color: rgba(115, 34, 93, 0.66);
    border: 2px solid rgb(255, 255, 255);
  }

  .container {
    display: grid;
    grid-template-columns: repeat(5, 0.4fr);
    justify-content: center;
    align-content: center;
  }

  ul {
    list-style: none;
  }
}
