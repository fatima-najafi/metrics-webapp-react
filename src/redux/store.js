import { configureStore } from '@reduxjs/toolkit';
import logger from 'redux-logger';
import companiesReducer from './companySlice';

const store = configureStore({
  reducer: {
    companies: companiesReducer,
  },
  middleware: (getDefaultMiddleware) => getDefaultMiddleware().concat(logger),
});

export default store;
