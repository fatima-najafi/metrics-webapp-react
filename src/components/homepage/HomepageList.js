import React, { useEffect, useState } from 'react';
import { v4 as uuidv4 } from 'uuid';
import { useNavigate } from 'react-router-dom';
import { useSelector, useDispatch } from 'react-redux';
import { getCompanies } from '../../redux/companySlice';
import HomepageItem from './HomepageItem';
import './HomepageList.css';

const HomepageList = () => {
  const navigate = useNavigate();
  const [search, setSearch] = useState('');
  const { companyList, status, error } = useSelector(
    (state) => state.companies,
  );
  const dispatch = useDispatch();

  useEffect(() => {
    dispatch(getCompanies());
  }, [dispatch]);

  if (status === 'loading') {
    return <div>Loading...</div>;
  }

  if (status === 'failed') {
    return (
      <div>
        Error:
        {' '}
        {error}
      </div>
    );
  }

  const filteredCompanies = Array.isArray(companyList)
    ? companyList.filter((company) => company.symbol.toLowerCase().includes(search.toLowerCase()))
    : [];

  return (
    <div className="context">
      <div className="search">
        <input
          className="input"
          type="text"
          onChange={(e) => setSearch(e.target.value)}
          placeholder="Search Company"
          value={search}
        />
      </div>

      <div className="container">
        {filteredCompanies.map((company) => (
          <div
            className="style"
            key={uuidv4()}
            onClick={() => navigate(`/details/${company.symbol}`)}
            aria-hidden="true"
          >
            <HomepageItem title={company.symbol} />
          </div>
        ))}
      </div>
    </div>
  );
};

export default HomepageList;
