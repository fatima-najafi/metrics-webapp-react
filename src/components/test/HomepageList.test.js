import React from 'react';
import { render } from '@testing-library/react';
import { BrowserRouter } from 'react-router-dom';
import { Provider } from 'react-redux';
import store from '../../redux/store';
import HomepageList from '../homepage/HomepageList';

describe('HomepageList', () => {
  test('renders correctly', () => {
    const { container } = render(
      <Provider store={store}>
        <BrowserRouter>
          <HomepageList />
        </BrowserRouter>
      </Provider>,
    );

    expect(container).toMatchSnapshot();
  });
});
