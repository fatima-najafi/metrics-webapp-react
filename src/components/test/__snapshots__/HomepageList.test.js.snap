// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HomepageList renders correctly 1`] = `
<div>
  <div>
    Loading...
  </div>
</div>
`;
